src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
+        method: "GET",
+        headers: { accept: "application/json" }
```

Comment:
Since the value of `headers` never changes, you could pull it out of `fetchJokes` and just declare it once outside of the component. Every time `fetchJokes` is called, it creates a new object. Since it's value never changes, it's safe to keep reusing the same one. Every time you create a new object, it uses memory that will need to be garbage collected eventually. 

For something this small, it really doesn't matter, but I'm trying to help you learn beyond this code 😅